src/client/src/spu/spu_pool.rs

```diff
@@ -159,6 +159,16 @@ impl SpuPool {
         Ok(stream)
     }
 
+    /// Returns true if key `topic/0` is found
+    pub async fn topic_exists<S: Into<String>>(&self, topic: S) -> bool {
+        let replica = ReplicaKey::new(topic, 0);
+        self.metadata
+            .partitions()
+            .lookup_by_key(&replica)
```

Comment:
You are right.  `lookup_by_key` potentially waits for while.  

You can do this to remove wait:

```
        self.metadata
            .partitions()
            .store()
            .contains_key(&replica)
            .await
``

Reply:
Cool. That performs a lot better now.

```
$ time target/release/fluvio produce nontopic
Error: 
   0: Consumer Error
   1: Fluvio client error
   2: Topic not found: nontopic

real    0m0.015s
user    0m0.007s
sys     0m0.014s
```

Reply:
Nice!

Rebase and we can merge this